File: WEEK2/bj8_1260.py
import sys
from collections import deque


def dfs(node):
    visited[node] = True
    dfs_result.append(node)
    for neighbor in adj_list[node]:
        if not visited[neighbor]:
            dfs(neighbor)


def bfs(start_node):
    queue = deque([start_node])
    visited[start_node] = True

    while queue:
        node = queue.popleft()
        bfs_result.append(node)
        for neighbor in adj_list[node]:
            if not visited[neighbor]:
                visited[neighbor] = True
                queue.append(neighbor)


n, m, v = map(int, sys.stdin.readline().split())

adj_list = [[] for _ in range(n + 1)]
for _ in range(m):
    u, w = map(int, sys.stdin.readline().split())
    adj_list[u].append(w)
    adj_list[w].append(u)

for i in range(1, n + 1):
    adj_list[i].sort()

visited = [False] * (n + 1)
dfs_result = []
dfs(v)
print(*dfs_result)

visited = [False] * (n + 1)
bfs_result = []
bfs(v)
print(*bfs_result)

File: WEEK2/bj8_1260.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const [n, m, v] = input[0].split(' ').map(Number);
const edges = input.slice(1);

const adjList = Array.from({ length: n + 1 }, () => []);
for (let i = 0; i < m; i++) {
    const [u, w] = edges[i].split(' ').map(Number);
    adjList[u].push(w);
    adjList[w].push(u);
}

for (let i = 1; i <= n; i++) {
    adjList[i].sort((a, b) => a - b);
}

let visited = new Array(n + 1).fill(false);
const dfsResult = [];
function dfs(node) {
    visited[node] = true;
    dfsResult.push(node);

    for (const neighbor of adjList[node]) {
        if (!visited[neighbor]) {
            dfs(neighbor);
        }
    }
}

dfs(v);
console.log(dfsResult.join(' '));

visited = new Array(n + 1).fill(false);
const bfsResult = [];
function bfs(startNode) {
    const queue = [startNode];
    visited[startNode] = true;

    while (queue.length > 0) {
        const node = queue.shift();
        bfsResult.push(node);

        for (const neighbor of adjList[node]) {
            if (!visited[neighbor]) {
                visited[neighbor] = true;
                queue.push(neighbor);
            }
        }
    }
}

bfs(v);
console.log(bfsResult.join(' '));
